File: prExplorer/Domain/Product.cs
namespace Domain
{
    public class Product
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public decimal DiscountPercentage { get; set; }
        public decimal Rating { get; set; }
        public int Stock { get; set; }
        public string Brand { get; set; }
        public string Thumbnail { get; set; }
        public string AppUserId { get; set; }
        public Guid CategoryId { get; set; }
        public AppUser AppUser { get; set; }
        public Category Category { get; set; }
    }
}

File: prExplorer/client-app/src/app/models/product.ts
export interface Product {
  id: string; // [CR 19-08-2023] It would be useful to have a Guid type for IDs in your code. Think how could this be realized
  title: string;
  description: string;
  price: number;
  discountPercentage: number;
  rating: number;
  stock: number;
  brand: string;
  category: string;
  thumbnail: string;
}
